### PR TITLE
[androidtv] Fixes Bugs and Prepares for PhilipsTV

### LINKED ISSUE #15631: [androidtv] Detection of Protocol v1
Devices which haven't been updated to version 2 of the protocol do not throw any visible errors when PIN is failing.  However, this message is sent and needs to be watched for so an error can be thrown to the user to update.

```
2023-09-19 15:21:32.862 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - Raw GoogleTV command decodes as: 34080210c801522d0a19636f6d2e676f6f676c652e616e64726f69642e766964656f73121073616d73756e6720534d2d4739393855
2023-09-19 15:21:32.862 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - Raw GoogleTV command decodes as: 16080210c801a2010e0a04080310060a04080310041802
2023-09-19 15:21:32.864 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - readerThreadJob message length 0
2023-09-19 15:21:32.864 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - readerThreadJob message length 0
2023-09-19 15:21:32.864 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - readerThreadJob message length 0
2023-09-19 15:21:32.864 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - readerThreadJob message length 35
2023-09-19 15:21:32.864 [TRACE] [l.googletv.GoogleTVConnectionManager] - livingroom - GoogleTV Message: 35 237b2270726f746f636f6c5f76657273696f6e223a312c22737461747573223a3430307d
2023-09-19 15:21:32.864 [TRACE] [tocol.googletv.GoogleTVMessageParser] - livingroom - Received GoogleTV message - Length: 35 Message: 7b2270726f746f636f6c5f76657273696f6e223a312c22737461747573223a3430307d
2023-09-19 15:21:32.864 [INFO ] [tocol.googletv.GoogleTVMessageParser] - livingroom - Unknown payload received. 35 7b2270726f746f636f6c5f76657273696f6e223a312c22737461747573223a3430307d

```

HEX decodes to 
`{"protocol_version":1,"status":400}`

### LINKED ISSUE #15632: [androidtv] ShieldTV Protocol Exception when App Updates
When an app updates, ShieldTV sends the following (example) which causes an exception.  This should be handled differently.

```
2023-09-19 15:24:11.071 [TRACE] [tocol.shieldtv.ShieldTVMessageParser] - livingroom - Received ShieldTV message from: SHIELD - Message: 08f107122a080512260a24636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e7365727669636518f107
2023-09-19 15:24:11.071 [INFO ] [tocol.shieldtv.ShieldTVMessageParser] - livingroom - Individual App Update - Please Report This: 08f107122a080512260a24636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e7365727669636518f107
2023-09-19 15:24:12.152 [TRACE] [tocol.shieldtv.ShieldTVMessageParser] - livingroom - Received ShieldTV message from: SHIELD - Message: 08f10712870108041282010a24636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e736572766963651219416e64726f69642054562052656d6f74652053657276696365223d687474703a2f2f31302e3235352e302e32303a33363931372f636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e73657276696365280118f107
2023-09-19 15:24:12.152 [WARN ] [tocol.shieldtv.ShieldTVMessageParser] - livingroom - Message Parser Exception on 08f10712870108041282010a24636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e736572766963651219416e64726f69642054562052656d6f74652053657276696365223d687474703a2f2f31302e3235352e302e32303a33363931372f636f6d2e676f6f676c652e616e64726f69642e74762e72656d6f74652e73657276696365280118f107
2023-09-19 15:24:12.153 [WARN ] [tocol.shieldtv.ShieldTVMessageParser] - livingroom - Message Parser Caught Exception

```

### LINKED ISSUE #16198: [androidtv] Setting port thing configuration on ShieldTV causes protocol failure
ShieldTV uses both GTV and STV protocols.  Setting port to a non default under the thing configuration causes it to be set for both protocols.  This will cause one protocol to fail as they will both try to connect to the same port.  This should be split per protocol.

### PR DESCRIPTION
**Pull request by morph166955**
Updates binding structure and adds basis for PhilipsTV additions in #15352 

Resolves #15632 
Resolves #15631 
Resolves #16198

### CODE REVIEW

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -43,9 +43,15 @@
 				<label>Hostname</label>
 				<description>Hostname or IP address of the device</description>
 			</parameter>
-			<parameter name="port" type="integer">
-				<label>Port</label>
+			<parameter name="googletvPort" type="integer">
```
Comment (lsiepel) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:46]: Changing the parameter name is a breaking change. Is this really that important? If so:

1.  a breaking change alert should be added to the distro repo
2.  readme has to be adapted
3. i18n plugin has to be run to generate translation for defautl language

Same for the `googletvPort` parameter.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVDynamicStateDescriptionProvider.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ * @author Ben Rosenblum - Initial contribution
+ */
+@Component(service = { DynamicStateDescriptionProvider.class,
+        AndroidTVDynamicStateDescriptionProvider.class }, immediate = true)
+@NonNullByDefault
+public class AndroidTVDynamicStateDescriptionProvider implements DynamicStateDescriptionProvider {
+    private final Map<ChannelUID, List<@NonNull StateOption>> channelOptionsMap = new ConcurrentHashMap<>();
+
+    // @SuppressWarnings("null")
```
Comment (lsiepel) [inline bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVDynamicStateDescriptionProvider.java:43]: please remove comment

**Review by lsiepel — CHANGES_REQUESTED** [review]
Thanks, left some comments/questions.

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:46]: Unfortunately yes this is required and also fixes a bug. If the port is set on a Shield (or eventually a Philips) that is not default it will use it for both protocols this breaking at least one.   I'll update as requested.

**Comment by morph166955** [comment]
@lsiepel Updates pushed.  https://github.com/openhab/openhab-distro/pull/1625 opened for distro.

**Review by lsiepel — APPROVED** [review]
LGTM

**Comment by morph166955** [comment]
Thank you @lsiepel !

**Comment by morph166955** [comment]
I pushed one small update to still catch the individual app updates and just dump them to debug.  If not we would have fallen through and spit an error message.

**Comment by morph166955** [comment]
Added a fix for #16223 as well.

**Comment by morph166955** [comment]
@wborn @jlaur Could I trouble one of you to give a quick look and merge this in?  @lsiepel has gone through it already and I've been running these changes on my system for a while now with no issues (they are included in the marketplace build).  I need to rebase #15352 so it can be reviewed and I have a bunch of other changes I want to submit which will rely on this being merged in.  Thank  you!

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:46]: Can this parameter be marked as advanced if the default port will work for most users?

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -43,9 +43,15 @@
...
+				<label>GoogleTV Port</label>
 				<description>Port to connect to</description>
+				<default>6466</default>
+			</parameter>
+			<parameter name="shieldtvPort" type="integer">
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:51]: Can this parameter be marked as advanced if the default port will work for most users?

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -114,9 +120,10 @@
 				<label>Hostname</label>
 				<description>Hostname or IP address of the device</description>
 			</parameter>
-			<parameter name="port" type="integer">
-				<label>Port</label>
+			<parameter name="googletvPort" type="integer">
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:123]: Can this parameter be marked as advanced if the default port will work for most users?

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVBindingConstants.java [hunk 5]
```diff
@@ -50,6 +50,8 @@ public class AndroidTVBindingConstants {
 
     // List of all config properties
     public static final String PROPERTY_IP_ADDRESS = "ipAddress";
+    public static final String PROPERTY_GOOGLETVPORT = "googletvPort";
+    public static final String PROPERTY_SHIELDTVPORT = "shieldtvPort";
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVBindingConstants.java:54]: Naming: These are parameters, not properties.

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml (hunk 4 above)
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:123]: Is it necesarry to rename this parameter and by that breaking backwards compatibility? There is still only one port parameter for this thing.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java [hunk 6]
```diff
@@ -89,6 +99,22 @@ public String getThingID() {
         return this.thingID;
     }
 
+    public Configuration getThingConfig() {
+        return getConfig();
+    }
+
+    public DiscoveryServiceRegistry getDiscoveryServiceRegistry() {
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:106]: I didn't load this into my IDE yet, but where is this method used?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the fixes. I have added a few comments from a quick look. I will try to have another look in the evening and make sure I didn't miss anything.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java (hunk 6 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:106]: It's in the PhilipsTV PR but will also likely be used when I do the Sony Scalar addition shortly as well.  I wanted to get it in now so I didn't have it sitting in both.  This PR provides a few of those small things.  It's called in PR #15352 go to PhilipsTVConnectionManager line 142.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVBindingConstants.java (hunk 5 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVBindingConstants.java]: I updated all 4 in there that are actually parameters.

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml (hunk 4 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:125]: I suppose we don't actually "have to" but I did for code clarity purposes.  Otherwise I'm going to end up having "port", "shieldtvPort", "philipstvPort", etc all in one file and it could be confusing as to which port "port" is really tied to. 
 Reality is, with ATV being a newer binding, and because the protocol stacks come from Google (and there is no option to change it on the device), I doubt anyone has actually configured this out of default.  I considered not even including a port option but did on the off chance someone had a weird NAT setup or something.  I'm not worried about breaking backwards compatibility, I really don't think anyone has actually ever used it.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java (hunk 6 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:106]: It's in the PhilipsTV PR but will also likely be used when I do the Sony Scalar addition shortly as well. I wanted to get it in now so I didn't have it sitting in both. This PR provides a few of those small things. It's called in PR https://github.com/openhab/openhab-addons/pull/15352 go to PhilipsTVConnectionManager line 142.

**Comment by morph166955** [comment]
@jlaur I've pushed updates for what you noted in the review.  I also marked several other parameters as advanced while I was in there.  Let me know what you would like to do with the two unresolved items above.  Thank you for the review!

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java (hunk 6 above)
Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:106]: I'm not sure if you will actually need it. You could have a look at #16250 although I'm not sure if it's a similar use-case. Maybe you find inspiration, maybe not. I cannot fully assess it right now, since it would require a deeper look at that PR. 

In any case it would be better to remove it from this PR since it's still unused. If it will ultimately be needed, you can introduce it with the first PR requiring it.

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml (hunk 4 above)
Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/thing/thing-types.xml:131]: OK, let's leave it then.

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/i18n/androidtv.properties [hunk 7]
```diff
@@ -84,5 +86,16 @@ offline.interrupted = Interrupted
...
+offline.error-occured-while-presenting-pairing-code = Error occurred while trying to present a Pairing Code on TV
+offline.error-occured-during-retrieval-of-credentials = Error occurred during retrieval of credentials
+offline.pairing-is-not-configured-yet = Pairing is not configured yet
+offline.error-occurred-while-trying-to-present-a-pairing-code-on-tv = Error occurred while trying to present a Pairing Code on TV
+offline.pairing-code-is-available-but-credentials-missing = Pairing Code is available, but credentials missing. Trying to retrieve them
+offline.error-occurred-during-retrieval-of-credentials = Error occurred during retrieval of credentials
+offline.pairing-was-unsuccessful = Pairing was unsuccessful
+offline.error-occurred-during-creation-of-http-client = Error occurred during creation of HTTP client
+offline.authentication-with-philips-tv-device-was-successful-continuing-initialization-of-the-tv = Authentication with Philips TV device was successful. Continuing initialization of the tv.
+offline.could-not-successfully-finish-pairing-process-with-the-tv = Could not successfully finish pairing process with the TV
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/i18n/androidtv.properties:98]: It seems that these are prematurely introduced, would be better to revert and only introduce with the PR using them.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java (hunk 6 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:106]: Just to be clear, are you wanting me to pull just this function or the whole thing out of the factory and all?  I don't have a problem with either, just making sure I understand the request.

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:106]: Yes, I would leave everything out until it will eventually be needed.

It was a good idea to extract the bugfixes from #15352 so they can be merged faster. You could even have extracted them individually, and it could have been even smoother, and each one could be mentioned individually in the release notes generated from the list of PR's. 😉

It has pros and cons, of course, but usually I think the pros outweights the cons. When increments are small, the risk of conflicts is usually also smaller. And since the reviews and merges will usually also be faster, everything can be rebased gradually, and there will be fewer and fewer still pending differences to manage. Additionally, it will be easier (well, possible) to cherry-pick commits for backporting specific bugfixes into patch releases.

File: bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/i18n/androidtv.properties [hunk 8]
```diff
@@ -85,4 +87,5 @@ offline.io-error = I/O Error
 offline.runtime-exception = Runtime exception
 offline.user-forced-pin-process = User Forced PIN Process
 offline.unknown = Unknown
+online.standby = Standby
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/resources/OH-INF/i18n/androidtv.properties:90]: I think this one also belongs to the other PR?

**Comment by morph166955** [comment]
Ok @jlaur I think that does it...

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java [hunk 9]
```diff
@@ -89,6 +94,14 @@ public String getThingID() {
         return this.thingID;
     }
 
+    public AndroidTVDynamicStateDescriptionProvider getStateDescriptionProvider() {
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:97]: Unfortunately I didn't spot this in my quick review earlier today, but this is not used either - see https://github.com/openhab/openhab-addons/pull/16191#discussion_r1448719145 - and should be removed as well.

 I now understand what you mean by "Prepares for PhilipsTV", but this is not the right way. You mentioned also Sony Scalar as well.

If you had only the PhilipsTV PR pending depending on this change, I would suggest to simply keep the change in that PR as you did originally. In case you will also need exactly the same ground work in your other upcoming PR, you could do the changes in a local branch and branch out to both branches from that, so that each branch would include it, but at the same time make it easier to manage separately (i.e. you could make changes in the original branch and rebase the other two against that). The first PR that gets merged, will introduce it, and the other one can be rebased then.

So in other words, I would keep this PR self-contained, i.e. probably strictly as a bugfix PR and not introduce any methods that are not currently used. It's very hard to review/assess the need without looking at the others PR's as well, and if they would never happen (for any reason) this code makes no sense.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java [hunk 10]
```diff
@@ -89,6 +94,14 @@ public String getThingID() {
...
+        return stateDescriptionProvider;
+    }
+
+    public ThingUID getThingUID() {
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:101]: Also unused.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java (hunk 9 above)
Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:97]: I see your point, I'll pull those out.  My problem with Philips and soon to be Sony is that they make changes to exactly the same lines of code so it's hard to maintain the two branches.  For example, in the thing factory I would be adding a thing type for both items.  They will end up on the same line in there.  Is it then your suggestion to create effectively 3 PRs?  One for Philips, one for Sony, and one with the "both need this" like the thing types?

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java]: My suggestion is to create only two PR's, but your two corresponding branches could be branched out from a common branch introducing this groundwork, i.e.:
- main
  - androidtv-preparations
    - androidtv-philips
    - androidtv-sony

When you create the PR's, each will drag the preparations along with them, but they will not depend on each other, so they won't have to merged in a particular order. When one of them is merged, you can rebase and resolve the few conflicts where lines are changed just next to each other. While you are working on both, you can still make changes directly to `androidtv-preparations` and rebase the other two against that to get the latest changes into each branch.

This is just some brainstorming on how I could imagine managing this situation. You might have other solutions/preferences, and that's fine as well. The important thing would be to try to keep things separate and self-contained.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java]: That makes sense.  I've pushed a commit to remove as requested.  An alternative (hope) in this case would be to just get the philips reviewed and merged in since it's "more or less" done.  At that point this isn't an issue as it would all be merged prior to Sony being submitted.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java [hunk 11]
```diff
@@ -159,13 +163,15 @@ public void initialize() {
         String ipAddress = googletvConfig.ipAddress;
         boolean gtvEnabled = googletvConfig.gtvEnabled;
 
-        if (ipAddress.isBlank()) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "@text/offline.googletv-address-not-specified");
-            return;
-        }
+        setThingProperty(PARAMETER_IP_ADDRESS, ipAddress);
```
Comment (jlaur) [inline bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Why do you need to have this as a property as well?

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: That had to be there to fix discovery.  We found the same on Roku recently.  By setting it there, the duplicate entry isn't populated in the inbox.

**Review by jlaur — COMMENTED** [review]
LGTM now, only one small question left when stumpled upon a parameter being duplicated into a property.

File: bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java (hunk 11 above)
Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: But the parameter is required, so it should be there even for manually created things. Can you share some more details about what is not working? I'm pretty sure this should work.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: In https://github.com/openhab/openhab-addons/pull/16210 we ended up having to add thing.setProperty(PROPERTY_UUID, deviceInfo.getSerialNumber().toLowerCase()); to RokuHandler.java for the same reason.  If the property is not set, then manually created devices (either GUI or text) are not recognized as the same thing, and thus not auto-ignored.  representation-property only checks the properties, not the parameters on the thing config.  In the case of ATV, there isn't a serial number or anything else that's unique to set to, especially when looking across the protocols.  The only thing I have is the IP of the device.  In this PR we modify the discovery participants to add .withRepresentationProperty(PARAMETER_IP_ADDRESS).withProperty(PARAMETER_IP_ADDRESS, ipAddress).  Reality is, we're just using the same name of "ipAddress" as both a parameter and a property.

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Can you try to add the representation property to the thing definition? See #12217.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: It's already there. Has been since the original merge.

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Sorry, I was on the phone and didn't see it. You are right. However, the case seems very similar to Danfoss Air Unit, and here it works. Definition:
https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.danfossairunit/src/main/resources/OH-INF/thing/thing-types.xml#L23-L29

Discovery:
https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/discovery/DanfossAirUnitDiscoveryService.java#L136-L146

And nothing in the handler except using the parameter. The host doesn't exist as a property. When I delete the ignored entry from the inbox and run a new scan, it's again added and automatically ignored:

image

I can try to have a look later what's going on since I also have an AndroidTV.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Just to be clear,  with Danfoss does it also ignore the result when you create the thing manually text and UI?  That's my problem here.

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Yes,, this is my thing:
```java
Thing danfossairunit:airunit:a2 "Danfoss HRV" [host="192.168.0.234",
refreshInterval=10,
updateUnchangedValuesEveryMillis=60000]
```

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Interesting.  I wonder why it works there but ATV and Roku did not.  I'll go grab some screen shots as examples.

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: I assume you have configured exactly the same IP address as discovered and not using a hostname?

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Apparently screenshots are harder than I expected.  The property isn't unesetting from the database when I install the addon from the current snapshot as opposed to the test jar.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > Yes,, this is my thing:
> 
> ```java
> Thing danfossairunit:airunit:a2 "Danfoss HRV" [host="192.168.0.234",
> refreshInterval=10,
> updateUnchangedValuesEveryMillis=60000]
> ```

When you go into the thing in the UI, do you see a host property set as well?

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > I assume you have configured exactly the same IP address as discovered and not using a hostname?

Yes, identical IPs

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > When you go into the thing in the UI, do you see a host property set as well?

No.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: The host property being set here:

properties

Makes these entries ignored:

inbox

To note, there is actually another issue I'm not sure how to resolve with this.  Notice the googletv entries don't auto-ignore but the shieldtv do.  I'm not sure how to make the googletv's tie to the shieldtvs existing.  I'm assuming this would have to be added to core to work.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > No.

Interesting.  Not sure why it works on some and not on others then.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Ok, I was able to replicate this.

I started with the binding as it exists in the snapshot I'm on (3802 for reference).  I created a thing manually in the UI for my test system.  Notice, no host property is set here.  Also, this is just a GTV not STV thing.

gtv-commit

The inbox does not ignore the device.

discover-commit

I then swapped in the jar.  I did not remove the thing or remove the entry from the inbox.  Notice the host property is added.

gtv-jar

I then rescanned ATV and voila the entry is now ignored.

discover-jar

I'm not entirely sure of the why, because I'm not that familiar with the code for discovery in the core, but by setting ipAddress as a property it resolves the issue.

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > To note, there is actually another issue I'm not sure how to resolve with this. Notice the googletv entries don't auto-ignore but the shieldtv do. I'm not sure how to make the googletv's tie to the shieldtvs existing. I'm assuming this would have to be added to core to work.

You could try to return only the thing type UID for ShieldTV here:
https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/discovery/ShieldTVDiscoveryParticipant.java#L46-L49

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > I then rescanned ATV and voila the entry is now ignored.

With the snapshot version of the binding, did you also try to delete entries from the inbox to have a clean slate, and then perform a rescan?

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Also, as a comparison.  Roku creates a property called uuid which effectively the serial number in lower case and adds that as a property both in discovery and in the handler.  The only difference here is that uuid isn't a thing parameter.  I'm not sure if it matters that I've reused ipAddress and they created uuid or not.  Either way, it's the correlation to uuid that's making it ignore.  To test that originally I created a branch and only updated that one single line and it resolved the issue.  See https://github.com/openhab/openhab-addons/issues/16164 for that info.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > > To note, there is actually another issue I'm not sure how to resolve with this. Notice the googletv entries don't auto-ignore but the shieldtv do. I'm not sure how to make the googletv's tie to the shieldtvs existing. I'm assuming this would have to be added to core to work.
> 
> You could try to return only the thing type UID for ShieldTV here:
> 
> https://github.com/openhab/openhab-addons/blob/444d824a41b9d0e517c35e2736e59ad3aabcbeb8/bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/discovery/ShieldTVDiscoveryParticipant.java#L46-L49

I will give that a shot and submit a PR if I can get it working.  Thank you!

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > > I then rescanned ATV and voila the entry is now ignored.
> 
> With the snapshot version of the binding, did you also try to delete entries from the inbox to have a clean slate, and then perform a rescan?
> 

Yes I deleted the entry in the inbox and rescanned to get that first result above (the unignored entry) just to be sure.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: For uniformity in testing, this is the jar I'm currently running.  

https://github.com/morph166955/openhab-addons/releases/download/androidtv-beta/org.openhab.binding.androidtv-4.2.0-SNAPSHOT.jar

Comment (jlaur) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: > I will give that a shot and submit a PR if I can get it working. Thank you!

Going back to my comment here https://github.com/openhab/openhab-addons/pull/16191#discussion_r1449337947 - perhaps you could do a new PR for both discovery issues and postpone resolving #16223 until we have this figured out? Then this PR can be merged now.

Comment (morph166955) [reply bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java:166]: Sure. Gimme an hour or two to get to my box and I'll split it apart.

**Review by jlaur — APPROVED** [review]

**Comment by jlaur** [comment]
@morph166955 - thanks! Can you update the PR title so that it will make more sense when reading release notes?

**Comment by morph166955** [comment]
> @morph166955 - thanks! Can you update the PR title so that it will make more sense when reading release notes?

Done!  #16264 is now open and rebased so we can continue on the discovery issue
